src/Hal/Bundle/PhpMetricsCollector/Collector/PhpMetricsCollector.php

```diff
@@ -1,101 +1,122 @@
+            );
         });
-
-        // Prepare datas
-        $all = $average = array(
-            'cfiles' => 0,
-            'maintainability' => array(),
-            'commentWeight' => array(),
-            'complexity' => array(),
-            'loc' => array(),
-            'lloc' => array(),
-            'cloc' => array(),
-            'bugs' => array(),
-            'difficulty' => array(),
-            'intelligentContent' => array(),
-            'vocabulary' => array(),
-        );
-        $scoreByFile = array();
-
-        // group files into tmp folder
-        $folder = sys_get_temp_dir() . DIRECTORY_SEPARATOR . uniqid();
-        mkdir($folder);
-        foreach ($files as $file) {
-            // run PhpMetrics
-            $tokenizer = new Tokenizer();
-            $tokenType = new TokenType();
-
-            // halstead
-            $halstead = new Halstead($tokenizer, $tokenType);
-            $rHalstead = $halstead->calculate($file);
-
-            // loc
-            $loc = new Loc($tokenizer);
-            $rLoc = $loc->calculate($file);
-
-            // complexity
-            $complexity = new McCabe($tokenizer);
-            $rComplexity = $complexity->calculate($file);
-
-            // maintainability
-            $maintainability = new MaintainabilityIndex();
-            $rMaintenability = $maintainability->calculate($rHalstead, $rLoc, $rComplexity);
-
-            // store result
-            $files[$file] = array();
-            $all['cfiles']++;
-            $all['maintainability'][] = $scoreByFile[$file]['maintainability'] = $rMaintenability->getMaintainabilityIndex();
-            $all['commentWeight'][] = $scoreByFile[$file]['commentWeight'] = $rMaintenability->getCommentWeight();
-            $all['complexity'][] = $scoreByFile[$file]['complexity'] = $rComplexity->getCyclomaticComplexityNumber();
-            $all['loc'][] = $scoreByFile[$file]['loc'] = $rLoc->getLoc();
-            $all['lloc'][] = $scoreByFile[$file]['lloc'] = $rLoc->getLogicalLoc();
-            $all['cloc'][] = $scoreByFile[$file]['cloc'] = $rLoc->getCommentLoc();
-            $all['bugs'][] = $scoreByFile[$file]['bugs'] = $rHalstead->getBugs();
-            $all['difficulty'][] = $scoreByFile[$file]['difficulty'] = $rHalstead->getDifficulty();
-            $all['intelligentContent'][] = $scoreByFile[$file]['intelligentContent'] = $rHalstead->getIntelligentContent();
-            $all['vocabulary'][] = $scoreByFile[$file]['vocabulary'] = $rHalstead->getVocabulary();
+        $config = new Config();
+        $config->set('exclude', $excludes);
+        $config->set('files', $files);
+        $metrics = (new Analyze($config, new TestOutput(), new Issuer(new TestOutput())))->run($files);
+        foreach ($metrics->all() as $each) {
+            $each->set('violations', new Violations());
+        }
+        $classMetrics = array_filter($metrics->all(), function (Metric $metric) {
```

Comment:
Same thing here about missing type-hints: 
```suggestion
        $classMetrics = array_filter($metrics->all(), static function (Metric $metric): bool {
```